### PR TITLE
ADTS-58 Analisys and Changes for ExportGraph

### PR DESCRIPTION
**Pull request by JoseZarmada**

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -274,6 +273,33 @@ class App extends Component {
...
+      eventService.publishError(exc);
+    }
+
+    if (data) {
+      const blob = new Blob([ JSON.stringify(data) ], { type: "application/json" });
+      const downloadUrl = URL.createObjectURL(blob);
+      let downloadLink = document.createElement("a");
+      downloadLink.setAttribute("href", downloadUrl);
+      downloadLink.setAttribute("download", "Datamodel");
+      downloadLink.style.visibility = 'hidden';
```
Comment (jamescarpinter) [inline client/src/App.js]: Should these be double quotes around this string?

File: client/src/App.js [hunk 2]
```diff
@@ -274,6 +273,33 @@ class App extends Component {
...
+      document.body.appendChild(downloadLink);
+      downloadLink.click();
+      document.body.removeChild(downloadLink);
+      setTimeout(() => URL.revokeObjectURL(downloadUrl), 100);
```
Comment (jamescarpinter) [inline client/src/App.js:297]: Why 100ms? What are we waiting for here?

Comment (JoseZarmada) [reply client/src/App.js:297]: https://developer.mozilla.org/en-US/docs/Web/API/URL/revokeObjectURL As far as I could see in the existing code (ExportComponent) and some examples in the Internet its to let the browser know not to keep the reference to the file any longer.

File: client/src/App.js (hunk 1 above)
Comment (JoseZarmada) [reply client/src/App.js]: Done

**Review by jamescarpinter — APPROVED** [review]

File: client/src/App.js [hunk 3]
```diff
@@ -274,6 +273,33 @@ class App extends Component {
...
+      exc.customMessage = "Error in exporting graph";
+      eventService.publishError(exc);
+    }
+
+    if (data) {
+      const blob = new Blob([ JSON.stringify(data) ], { type: "application/json" });
+      const downloadUrl = URL.createObjectURL(blob);
+      let downloadLink = document.createElement("a");
+      downloadLink.setAttribute("href", downloadUrl);
+      downloadLink.setAttribute("download", "Datamodel");
```
Comment (jamiewilbraham) [inline client/src/App.js]: Can we change this to be "ExportedGraph" instead of "Datamodel"?

**Review by jamiewilbraham — CHANGES_REQUESTED** [review]

**Pull request merged**